nntrainer/src/bn_layer.cpp

```diff
@@ -87,69 +87,71 @@ int BatchNormalizationLayer::setProperty(std::vector<std::string> values) {
 
 Tensor BatchNormalizationLayer::forwarding(Tensor in, int &status) {
   Tensor temp;
-  assert(dim.batch() > 0);
-  hidden = in;
+  this->input = in;
 
-  mu = in.sum(0).multiply(1.0 / dim.batch());
+  this->mu = in.average(0);
```

Comment:
It is not relevant this PR but I think the implementation of this average (int axis) is something wrong. It looks like only works for axis 0 (batch). cause it divides with batch not axis. Could you check it as well?

Reply:
Yes. Thanks for pointing out. I made an issue for that 👍